enricher/api/src/main/java/io/fabric8/maven/enricher/api/Enricher.java

```diff
@@ -63,14 +64,14 @@
      *
      * @param builder the build to examine and add to
      */
-    void addMissingResources(KubernetesListBuilder builder);
+    void addMissingResources(PlatformMode platformMode, KubernetesListBuilder builder);
```

Comment:
The idea is the right one, but I'm missing the full implementation of #678 , i.e. removal of getLabels(), getAnnotations() and getSelector(), and moving to a create() and adapt() phase (as  described in #678). Is this something planned for later ?